webServices/workSpaceComponentDirectory/webhook.js

```diff
@@ -0,0 +1,60 @@
+"use strict";
+module.exports = {
+  // Generic components data
+  type: 'Webhook',
+  description: 'Envoyer les données en POST sur une URL externe.',
+  editor: 'webhook-editor',
+  graphIcon: 'Webhook.png',
+  tags: [
+    'http://semantic-bus.org/data/tags/outComponents',
+    'http://semantic-bus.org/data/tags/APIComponents'
+  ],
+  // Component-specific data
+  fetch: require('node-fetch'),
+
+  // Function called when another component push to this component
+  pull: function(data, flowData, queryParams) {
+    const componentConfig = data.specificData;
+    let body;
+
+    switch (componentConfig.contentType) {
+      case 'application/json':
+        body = JSON.stringify(flowData[0].data);
+        break;
+
+      default:
+        return Promise.reject(new Error('Only application/json contentType is currently supported for Webhook component'));
+    }
+
+    return this.call_webhook(componentConfig.url, {
+        method: 'POST',
+        body: body,
+        headers: {'Content-Type': componentConfig.contentType}
+    });
+  },
+
+  call_webhook: function(url, options, numRetry) {
+      if( numRetry === undefined ) numRetry = 0;
+
+      return this.fetch(url, options).catch(error => {
+          if( numRetry >= 7 ) {
+              // TODO log the failed webhook posts somewhere ?
+              console.error(error);
+          } else {
+              // Exponentially increment retry interval at every failure
+              // This will retry after 5s, 25s, 2m, 10m, 50m, 4h, 21h
+              const retryInterval = Math.pow(5, numRetry+1);
+
+              console.log(`Webhook post to ${url} failed, trying again in ${retryInterval}s...`);
```

Comment:
Yes that's the idea. The URL to call may be down temporarly, IMO this shouldn't be an error.
Do you think we should do it differently ? Should I remove the console.log ?

Reply:
The log is valid, it is the level I would level up to *warning* at least.
Maybe not the first times, but if the URL is down 1h (or even 30m), it is already long enough to be informed and maybe to check the issue.

What about this compromise:

* if numRetry < 4 (for 10m), console.log (like today)
* else if numRetry > 7 (for 21h), console.error and stop (like today)
* else (for 10m, 50m and 4h) console.warn (new case)

It could also be a more configurable/smart strategy, I mainly give the idea.